frontend/src/modules/LoginRedux.js

```diff
@@ -8,8 +10,22 @@ export const IS_CHANGING_PASSWORD = "IS_CHANGING_PASSWORD";
 export const CHANGE_PASSWORD_SUCCESS = "CHANGE_PASSWORD_SUCCESS";
 export const CHANGE_PASSWORD_FAILURE = "CHANGE_PASSWORD_FAILURE";
 
-// Simplified version of the authentication action (temporary)
-const authenticateAction = authenticated => ({ type: AUTHENTICATED, authenticated });
+function authenticateAction(response, dispatch, location, push) {
+  return async function() {
+    if (navigator.cookieEnabled) {
+      localStorage.setItem("token", response.token);
+    }
+
+    if (location === PATH.login) {
+      push(PATH.dashboard);
+    }
+    return dispatch({ type: AUTHENTICATED });
+  };
+}
+
+function updateAuthenticatedState() {
```

Comment:
The difference between `updateAuthenticatedState` and `authenticateAction` is confusing. I think they need re-naming and some comments.

I think this should stay as is and be used whenever you need to update the state.
`const authenticateAction = authenticated => ({ type: AUTHENTICATED, authenticated });`

Then your more complex function that handles local storage should be named something more specific like `handleAuthResponseAndState` ??

Reply:
Refactored in commit #https://github.com/code-for-canada/project-thundercat/pull/312/commits/1226ba3db4c64a02aace8e13f5aef15bef0c565c